projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -680,6 +681,11 @@ export class IgxOverlayService implements OnDestroy {
         if (info.initialSize.width !== 0) {
             info.elementRef.nativeElement.parentElement.style.width = info.initialSize.width + 'px';
         }
+
+        // set content div size
```

Comment:
too late to set the `size` here. If we attach a component ref, e.g. `igxCalendar` in `igxDatePicker` on line 677 we will get the initial size without taking into account the its potential `--ig-size`. Then here we are setting the `--ig-size` and initial size may end up with wrong value. Move the setting of `--ig-size` in the very beginning of `updateSize` method.